cassandra/cassandra_session.go

```diff
@@ -0,0 +1,137 @@
+package cassandra
+
+import (
+	"sync"
+	"time"
+
+	"github.com/gocql/gocql"
+	log "github.com/sirupsen/logrus"
+)
+
+// Session stores a connection to Cassandra along with associated configurations
+type Session struct {
+	session                 *gocql.Session
+	cluster                 *gocql.ClusterConfig
+	shutdown                chan struct{}
+	connectionCheckTimeout  time.Duration
+	connectionCheckInterval time.Duration
+	addrs                   string
+	logPrefix               string
+	sync.RWMutex
+}
+
+// NewSession creates and returns a CassandraSession
+func NewSession(session *gocql.Session,
```

Comment:
if the `*Session` which gets returned from `NewSession()` has the ability to create cassandra connections already, why still require the user to pass one in? Shouldn't it be sufficient if the user just provides the `*gocql.ClusterConfig` and then nothing outside of this package needs to create `*gocql.Session` instances